### PR TITLE
Clearer logging for autocommisioning parameter setup

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Log message of 'no parameters available' is misleading: parameters are available however end device may not correctly report supported wifi/thread interfaces.

Ensure we log status of what parameters and capabilities are available.

#### Change overview
Slight code move-around and better logging.

#### Testing
Tested current m5stack which fails to provision, noticed the log of "device does not support any network setup"

### CODE REVIEW

File: src/controller/AutoCommissioner.cpp [hunk 1]
```diff
@@ -139,24 +141,47 @@ CommissioningStage AutoCommissioner::GetNextCommissioningStage(CommissioningStag
...
+                }
+                else
+                {
+                    ChipLogProgress(Controller, "WiFi parameters not available");
+                }
             }
-            else
+            if (mNetworkTechnology.Has(NetworkCommissioningFeature::kWiFiNetworkInterface))
             {
-                ChipLogError(Controller, "Required network information not provided in commissioning parameters");
```
Comment (cecille) [inline src/controller/AutoCommissioner.cpp:156]: This code section is being revised in https://github.com/project-chip/connectedhomeip/pull/14567, which obsolete this change fairly quickly. Instead, could this error message just be changed to report what commissioning parameters were supplied vs. what is supported in the cluster? 
Ex - Required network information not provided - %s parameters were supplied, device network commissioning cluster supports %s"

**Review by woody-apple — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
Fast tracking, this change has had enough time for review. (> 3 days)

File: src/controller/AutoCommissioner.cpp (hunk 1 above)
Comment (andy31415) [reply src/controller/AutoCommissioner.cpp:159]: I re-updated based on master.

**Comment by andy31415** [comment]
Closing: latest code seems to log yes/now on network technology stack support.

**Pull request closed without merging**